console/frontend/src/main/frontend/src/app/app.component.ts

```diff
@@ -676,4 +711,16 @@ export class AppComponent implements OnInit, OnDestroy {
       this.MODAL_OPTIONS_CLASSES,
     );
   }
+
+  private updateClusterMembers(
+    member: ClusterMember,
+    action: ClusterMemberEventType,
+  ): void {
+    // TODO If the server keeps sending events twice, check if id still/already exists
+    // On add only if id doesnt exist, on remove only if id exists
+    this.clusterMembers =
+      action === 'ADD_MEMBER'
+        ? [...this.clusterMembers, member]
+        : this.clusterMembers.filter((member) => member.id !== member.id);
```

Comment:
Eslint forced me to make it a tenary, so I'd have to ignore the line to make it not behave like that.
As for the addition of a new member, I dont want to mutate the variable but set it as a new object which is what the spread operator does shallowly, this works better for Angular since I'm using this property also in a child component.

Reply:
Shallow would juist niet be goed, toch? Maar I think you bedoelt deep.
Anyway, with an if-else of two ifs, it would feel safer.

Reply:
It _is_ shallow as in that the objects inside the array don't get recreated but the variable `clusterMembers` does get reassigned to a new object in memory, triggering angular to check for changes also in the child component where `clusterMembers` is being used

Reply:
Dan I had it verkeerd gelezen, but ik snap ook niet helemaal wat je bedoelt.